### PR TITLE
Use the schema to validate a credential being added

### PR DESCRIPTION
**Pull request by orhoj**
## Purpose
Actually use the schema to validate a credential being added. This will e.g. block giving a string for an integer attribute.

## Changes
- Validate credential with the JSON schema.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: packages/browser-wallet/src/popup/pages/AddWeb3IdCredential/AddWeb3IdCredential.tsx [hunk 1]
```diff
@@ -115,23 +116,31 @@ export default function AddWeb3IdCredential({ onAllow, onReject }: Props) {
...
-            const requiredAttributes = schema.properties.credentialSubject.properties.attributes.required;
-            for (const requiredAttribute of requiredAttributes) {
-                if (!Object.keys(credential.credentialSubject.attributes).includes(requiredAttribute)) {
-                    missingRequiredAttributeKeys.push(requiredAttribute);
+            // Use the schema to validate the credential.
+            const validator = new Validator();
+            try {
+                const validationResult = validator.validate(
+                    { credentialSubject: credential.credentialSubject },
+                    schema as unknown as Schema
```
Comment (shjortConcordium) [inline packages/browser-wallet/src/popup/pages/AddWeb3IdCredential/AddWeb3IdCredential.tsx:124]: This does allow me to do timestamp attributes as dates. Did we want allow that here or only in statements?

Comment (orhoj) [reply packages/browser-wallet/src/popup/pages/AddWeb3IdCredential/AddWeb3IdCredential.tsx:124]: I am not sure.

Comment (orhoj) [reply packages/browser-wallet/src/popup/pages/AddWeb3IdCredential/AddWeb3IdCredential.tsx:124]: Ah, I understand now. It does not allow it now. Also if I add the object type, then it is not displayed correctly (shows as [object Object].

Comment (orhoj) [reply packages/browser-wallet/src/popup/pages/AddWeb3IdCredential/AddWeb3IdCredential.tsx:124]: Going to as the core team about it. We will leave this out until we know.

Comment (orhoj) [reply packages/browser-wallet/src/popup/pages/AddWeb3IdCredential/AddWeb3IdCredential.tsx:124]: @shjortConcordium we decide how we want it. So I would be interested in your opinion. Restricting to the schema when adding would also fix the serialization issue we had (and already fixed).

**Review by shjortConcordium — APPROVED** [review]

**Pull request merged**